File: 246.cpp
class Solution {
public:
    bool isStrobogrammatic(string num) {
        int l = 0, r = num.size() - 1;
        int left, right;

        while(l <= r){
            left = num[l];
            right = num[r];
            if(left == '6') {
                left = '9';
            } else if(left == '9') {
                left = '6';
            } else if(left == '0' || left == '1' || left == '8'){
                left = left;
            } else {
                return false;
            }

            if(left != right) {
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
};

File: 246.py
class Solution:
    def isStrobogrammatic(self, num: str) -> bool:
        l, r = 0, len(num) - 1
        while(l <= r):
            left, right = int(num[l]), int(num[r])
            if(left in [2, 3, 4, 5, 7]):
                return False
            elif(left in [0, 1, 8]):
                left = left
            elif left == 6:
                left = 9
            else:
                left = 6
            if(left != right):
                return False
            l, r = l + 1, r - 1
        return True